spec/dockly/docker/ecr_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+describe Dockly::Docker::ECR do
+  subject do
+    described_class.new(name: 'ecr', server_address: registry)
+  end
+
+  let(:username) { 'AWS' }
+  let(:password) { 'password' }
+  let(:registry) { 'accountid.dkr.ecr.region.amazonaws.com' }
+  let(:endpoint) { "https://#{registry}" }
+  let(:ecr_client) { Aws::ECR::Client.new(stub_responses: true) }
+
+  after do
+    Aws.config[:ecr] = { stub_responses: {} }
+  end
+
+  describe '#authenticate!' do
+    context 'when AWS provides a username and password for ECR' do
+      before do
+        Aws.config[:ecr] = {
```

Comment:
Is this now the preferred way of doing this, rather than dependency injecting a client?

Reply:
I've recently found this approach flexible and easier to scale if we need to add more complex operations later, plus prefer to avoid stubbing when possible. Happy to discuss dependency injecting if we feel it'd be a better fit though

Reply:
Just trying to keep up with the times myself! Fewer arguments to objects always makes me feel better.